pkg/reconciler/vspheresource/vsphere.go

```diff
@@ -146,18 +146,17 @@ func (r *Reconciler) reconcileConfigMap(ctx context.Context, vms *sourcesv1alpha
 
 func (r *Reconciler) reconcileServiceAccount(ctx context.Context, vms *sourcesv1alpha1.VSphereSource) error {
 	ns := vms.Namespace
-	name := resourcenames.ServiceAccount(vms)
-
-	_, err := r.saLister.ServiceAccounts(ns).Get(name)
+	serviceAccountName := resourcenames.ServiceAccount(vms)
+	_, err := r.saLister.ServiceAccounts(ns).Get(serviceAccountName)
```

Comment:
In this case, if service account is empty, we shouldn't be attempting to fetch it.